libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -49,9 +49,8 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 	// stack layout: [source_ref] [source length] target_ref (top)
 	solAssert(_targetType.location() == DataLocation::Storage, "");
 
-	Type const* uint256 = TypeProvider::uint256();
-	Type const* targetBaseType = _targetType.isByteArrayOrString() ? uint256 : _targetType.baseType();
-	Type const* sourceBaseType = _sourceType.isByteArrayOrString() ? uint256 : _sourceType.baseType();
```

Comment:
@wechman Here, this original piece of code basically results in copying ``bytes`` as ``uint256[]``... so the logic below is no longer properly aware of the possibility of a tail that ought to be cleared or anything like that... to properly account for this, we'd probably basically have to rewrite the entire function specially for the ``bytes`` case instead of reusing the general array code - so we might as well just call the yul implementation... even though it may be slightly more expensive, but we want people to move to Yul IR codegen anyways, so I wouldn't spend too much effort on making this optimal for legacy codegen.